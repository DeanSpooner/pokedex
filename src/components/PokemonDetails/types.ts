export interface Props {
  pokemon: {
    name: string;
    base_experience: number;
    height: number;
    weight: number;
    types: [
      {
        type: {
          name: string;
        };
      }
    ];
  };
  id: string | string[];
}
