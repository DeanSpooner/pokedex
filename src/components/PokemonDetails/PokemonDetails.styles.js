import styled from "styled-components";

const handleColor = (elementType) => {
  switch (elementType) {
    case "bug":
      return "#57e853";
    case "dark":
      return "#181818;";
    case "dragon":
      return "#9f5ee1;";
    case "electric":
      return "#e6e330;";
    case "fairy":
      return "#da67e3;";
    case "fighting":
      return "#661012;";
    case "fire":
      return "#f8410a;";
    case "flying":
      return "#7ba4e3;";
    case "ghost":
      return "#140765;";
    case "grass":
      return "#184910;";
    case "ground":
      return "#e2a543;";
    case "ice":
      return "#24f4ea;";
    case "normal":
      return "#c0edee;";
    case "poison":
      return "#a100fd;";
    case "psychic":
      return "#e7095a;";
    case "rock":
      return "#974913;";
    case "steel":
      return "#272727;";
    case "water":
      return "#03a9f3;";
  }
};

export const PokemonDetailsWrapper = styled.div`
  text-align: center;
`;

export const PokemonImageWrapper = styled.div`
  border: solid 24px ${(props) => handleColor(props.firstType)};
  border-radius: 50%;
  margin: auto;
  padding: 32px;
  width: 240px;
`;

export const PokemonDetailText = styled.p`
  font-size: 24px;
  font-weight: 500;
`;
