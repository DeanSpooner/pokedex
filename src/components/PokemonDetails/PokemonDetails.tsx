import Image from "next/image";
import React from "react";
import PokemonType from "../PokemonType/PokemonType";
import {
  PokemonDetailsWrapper,
  PokemonImageWrapper,
  PokemonDetailText,
} from "./PokemonDetails.styles";
import { Props } from "./types";

const PokemonDetails: React.FC<Props> = ({ pokemon, id }) => {
  const capitalizeFirstLetter = (string: string) => {
    return string.charAt(0).toUpperCase() + string.slice(1);
  };

  return (
    <PokemonDetailsWrapper data-testid="PokemonDetails" className="">
      <h1>
        This is Pok√©mon #{id}, {capitalizeFirstLetter(pokemon?.name)}
      </h1>
      <PokemonImageWrapper firstType={pokemon?.types[0]?.type?.name}>
        <Image
          src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/${id}.png`}
          alt={`${capitalizeFirstLetter(pokemon?.name)} artwork`}
          width="240px"
          height="240px"
        />
      </PokemonImageWrapper>
      <PokemonDetailText>
        Types:{" "}
        {pokemon?.types?.map((type) => (
          <PokemonType elementType={type?.type?.name} key={type?.type?.name} />
        ))}
      </PokemonDetailText>
      <PokemonDetailText>
        Base experience: {pokemon?.base_experience}
      </PokemonDetailText>
      <PokemonDetailText>Height: {pokemon?.height / 10}m</PokemonDetailText>
      <PokemonDetailText>Weight: {pokemon?.weight / 10}kg</PokemonDetailText>
    </PokemonDetailsWrapper>
  );
};

export default PokemonDetails;
