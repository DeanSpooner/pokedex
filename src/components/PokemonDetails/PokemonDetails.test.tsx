/** @jest-environment jsdom */

import React from "react";
import { render } from "@testing-library/react";
import PokemonDetails from "./PokemonDetails";

describe("PokemonDetails", () => {
  it("should render with data", () => {
    PokemonDetails;
    const { getByTestId } = render(
      <PokemonDetails
        pokemon={{
          name: "Bulbasaur",
          base_experience: 64,
          types: [{ type: { name: "grass" } }],
          height: 7,
          weight: 69,
        }}
        id={"1"}
      />
    );
    expect(getByTestId("PokemonDetails")).toMatchSnapshot();
  });
});
