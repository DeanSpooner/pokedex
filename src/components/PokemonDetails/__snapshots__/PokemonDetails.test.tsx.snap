// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonDetails should render with data 1`] = `
<div
  class="PokemonDetailsstyles__PokemonDetailsWrapper-sc-vr58es-0 lcFgPG"
  data-testid="PokemonDetails"
>
  <h1>
    This is Pok√©mon #
    1
    , 
    Bulbasaur
  </h1>
  <div
    class="PokemonDetailsstyles__PokemonImageWrapper-sc-vr58es-1 flpqhd"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27240%27%20height=%27240%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="Bulbasaur artwork"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
  </div>
  <span
    class="PokemonDetailsstyles__PokemonDetailText-sc-vr58es-2 gPvLQO"
  >
    Types:
     
    <div
      class="PokemonTypestyles__MonsterType-sc-1b3gn2z-0 LISUX"
      data-testid="PokemonType"
    >
      GRASS
    </div>
  </span>
  <span
    class="PokemonDetailsstyles__PokemonDetailText-sc-vr58es-2 gPvLQO"
  >
    Base experience: 
    64
  </span>
  <span
    class="PokemonDetailsstyles__PokemonDetailText-sc-vr58es-2 gPvLQO"
  >
    Height: 
    0.7
    m
  </span>
  <span
    class="PokemonDetailsstyles__PokemonDetailText-sc-vr58es-2 gPvLQO"
  >
    Weight: 
    6.9
    kg
  </span>
</div>
`;
